File: src/main/java/com/zero_delusions/dev_assignment/command/OpenGuiCommand.java
package com.zero_delusions.dev_assignment.command;

import com.mojang.brigadier.CommandDispatcher;
import com.zero_delusions.dev_assignment.gui.CustomGuiScreenHandler;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;
import net.minecraft.network.chat.Component;
import net.minecraft.world.SimpleMenuProvider;

public final class OpenGuiCommand {
    public static void register(CommandDispatcher<CommandSourceStack> dispatcher) {
        dispatcher.register(
                Commands.literal("opengui")
                        .executes(context -> {
                            CommandSourceStack source = context.getSource();
                            var player = source.getPlayer();
                            if (player != null) {
                                player.openMenu(new SimpleMenuProvider(
                                        CustomGuiScreenHandler::new,
                                        Component.literal("Settings")
                                ));
                                return 1;
                            } else {
                                return 0;
                            }
                        })
        );
    }
}

File: src/main/kotlin/com/zero_delusions/dev_assignment/command/OpenGuiCommand.kt
package com.zero_delusions.dev_assignment.command

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.builder.LiteralArgumentBuilder.literal
import com.zero_delusions.dev_assignment.CustomGuiScreenHandler
import net.minecraft.commands.CommandSourceStack
import net.minecraft.network.chat.Component
import net.minecraft.world.SimpleMenuProvider

object OpenGuiCommand {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
            literal<CommandSourceStack>("opengui")
                .executes { context ->
                    val player = context.source.player
                    if (player != null) {
                        player.openMenu(SimpleMenuProvider(
                            { syncId, inventory, p ->
                                CustomGuiScreenHandler(syncId, inventory, p)
                            },
                            Component.literal("Settings")
                        ))
                        1
                    } else {
                        0
                    }
                }
        )
    }
}